coral/src/app/features/browse-topics/BrowseTopics.test.tsx

```diff
@@ -432,4 +433,123 @@ describe("BrowseTopics.tsx", () => {
       expect(getAllTopics()).toHaveLength(2);
     });
   });
+
+  describe("handles user searching by topic name with search input", () => {
+    const testSearchInput = "Searched for topic";
+    const getAllTopics = () =>
+      within(
+        screen.getByRole("table", { name: /Topics overview/ })
+      ).getAllByRole("rowheader");
+
+    beforeEach(() => {
+      mockTopicGetRequest({
+        mswInstance: server,
+      });
+      customRender(<BrowseTopics />, { memoryRouter: true, queryClient: true });
+    });
+
+    afterEach(() => {
+      server.resetHandlers();
+      cleanup();
+    });
+
+    it("fetches new data when user enters text in input and clicks the search button", async () => {
+      const input = screen.getByRole("searchbox", {
+        name: "Search by topic name",
+      });
+      const submitButton = screen.getByRole("button", {
+        name: "Submit search",
+      });
+
+      await waitForElementToBeRemoved(screen.getByText("Loading..."));
+
+      expect(getAllTopics()).toHaveLength(10);
+      expect(input).toHaveValue("");
+      expect(submitButton).toBeEnabled();
+
+      await userEvent.type(input, testSearchInput);
+
+      expect(input).toHaveValue(testSearchInput);
+
+      await userEvent.click(submitButton);
+      await waitForElementToBeRemoved(screen.getByText("Filtering list..."));
+
+      expect(getAllTopics()).toHaveLength(2);
```

Comment:
With this approach, we are putting a lot trust in our mocked API. I think it would be great, if we could somehow assert the API query parameters. As we are not in a terrible rush, could you look into if we could spy on [api.get](https://github.com/aiven/klaw/blob/main/coral/src/services/api.ts#L271) and then assert the arguments it was called with.

Reply:
That's a very good idea, and I was also slightly uneasy with this approach.

To be clear, what you are suggesting is :
- intercept the request
- assert if the query has params `search` with value `testSearchInput`

This seems like it would add a layer of certainty, for sure. I think that it might have its place in a subsequent PR, as the approach should probably be implemented in all tests which currently rely on the mocked API in the same way.

Reply:
I will merge this, as my requested change should only be addition.